lib/fauna/errors.rb

```diff
@@ -47,23 +43,8 @@ def self.raise_for_status_code(request_result)
     #            :: A simple string as the message.
     def initialize(request_result)
       @request_result = request_result
-
-      errors = request_result.response_content
-      if errors.is_a?(Hash)
-        if errors.key?(:error)
-          message = errors[:description] || errors[:error]
-          @errors = [message]
-        else
-          errors = errors[:errors]
-          message = errors.collect { |error| error[:code] }.join(',')
-          @errors = errors
-        end
-      else
-        message = errors
-        @errors = nil
-      end
-
-      super(message)
+      @errors = request_result.response_content[:errors].map(&ErrorData.method(:from_hash))
```

Comment:
We assume for safety `errors` might not exist. If nothing else, we can expect this to happen when a load balancer or reverse proxy gives out.